src/commands/send-many-memo-safe.ts

```diff
@@ -0,0 +1,284 @@
+import { Command, flags } from '@oclif/command';
+import { sendMany, Recipient, isNormalInteger, getAddress } from '../builder';
+import {
+  StacksMocknet,
+  StacksMainnet,
+  StacksTestnet,
+  StacksNetwork,
+} from '@stacks/network';
+import {
+  broadcastTransaction,
+  ChainID,
+  validateStacksAddress,
+} from '@stacks/transactions';
+import { STXPostCondition } from '@stacks/transactions/dist/transactions/src/postcondition';
+import fetch, { Response } from 'node-fetch';
+
+type NetworkString = 'mocknet' | 'mainnet' | 'testnet';
+
+const DEFAULT_TESTNET_CONTRACT =
+  'STR8P3RD1EHA8AA37ERSSSZSWKS9T2GYQFGXNA4C.send-many-memo';
+const DEFAULT_MAINNET_CONTRACT =
+  'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.send-many-memo';
+
+function uniqueFetchSession() {
+  return (() => {
+    let register: { [key: string]: Promise<Response> } = {};
+    return (url: string) => register[url] || (register[url] = fetch(url));
+  })();
+}
+
+async function checkMemoExpected(
+  network: StacksNetwork,
+  recipients: Recipient[]
+): Promise<string[]> {
+  const fetch = uniqueFetchSession();
+  const results = await Promise.all(
+    recipients
+      .filter(recipient => recipient.memo?.length === 0)
```

Comment:
I think that this will compile to `false` if the memo is undefined, because `undefined === 0` will be false. So, if the memo is `undefined`, this will not get caught, right?

Reply:
You're right, I wrongly assumed the memo would always be set since a split of `"a,b,"` becomes `["a","b",""]` but if the user leaves out the second comma then memo becomes `undefined`. Best to add the edge-case in the filter instead of defaulting to an empty string when splitting the recipients, I think.